sklearn/cluster/tests/test_k_means.py

```diff
@@ -93,19 +93,27 @@ def test_kmeans_relocated_clusters(array_constr, algo):
     # second center too far from others points will be empty at first iter
     init_centers = np.array([[0.5, 0.5], [3, 3]])
 
-    expected_labels = [0, 0, 1, 1]
-    expected_inertia = 0.25
-    expected_centers = [[0.25, 0], [0.75, 1]]
-    expected_n_iter = 3
-
     kmeans = KMeans(n_clusters=2, n_init=1, init=init_centers, algorithm=algo)
     kmeans.fit(X)
 
-    assert_array_equal(kmeans.labels_, expected_labels)
+    expected_n_iter = 3
+    expected_inertia = 0.25
     assert_allclose(kmeans.inertia_, expected_inertia)
-    assert_allclose(kmeans.cluster_centers_, expected_centers)
     assert kmeans.n_iter_ == expected_n_iter
 
+    # There are two acceptable ways of relocating clusters in this example, the output
+    # depends on how the argpartition strategy break ties. We accept both outputs.
+    try:
+        expected_labels = [0, 0, 1, 1]
+        expected_centers = [[0.25, 0], [0.75, 1]]
+        assert_array_equal(kmeans.labels_, expected_labels)
+        assert_allclose(kmeans.cluster_centers_, expected_centers)
+    except AssertionError:
+        expected_labels = [1, 1, 0, 0]
+        expected_centers = [[0.75, 1.0], [0.25, 0.0]]
+        assert_array_equal(kmeans.labels_, expected_labels)
+        assert_allclose(kmeans.cluster_centers_, expected_centers)
```

Comment:
maybe this is simpler ?
```suggestion
    expected_labels = [0, 0, 1, 1]
    expected_centers = [[0.25, 0], [0.75, 1]]
    assert_allclose(v_measure_score(kmeans.labels_, expected_labels), 1)
    assert_allclose(np.sort(kmeans.cluster_centers_, axis=0), expected_centers)
```

Reply:
The fact that we get the same centers but permuted is very particular to how the data is chosen at the beginning I think ? we could have a relocation issue with a tie that does not result in the same expected centers ? So I thought using this fact to make the assertion simpler is not very good.

Reply:
I thought that with the comment it was acceptable, but I'm fine with the current try except version